app/src/main/java/in/co/rajkumaar/amritarepo/timetable/AcademicTimetableActivity.java

```diff
@@ -1,303 +1,362 @@
+        course.setAdapter(courseAdapter);
+    }
+
+    private void setBranchSpinner() {
+        branchAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, branches);
+        branchAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        branch.setAdapter(branchAdapter);
+        if (course.getSelectedItemPosition() == pref.getInt("course", 0)) {
+            branch.setSelection(pref.getInt("branch", 0));
+        }
+    }
+
+    private void getCourse(final List<String> coursestemp) {
+        AsyncHttpClient client = new AsyncHttpClient();
+        client.get(getString(R.string.course_name_url), new AsyncHttpResponseHandler() {
+            @Override
+            public void onSuccess(int i, Header[] headers, byte[] bytes) {
+                Document html = Jsoup.parse(new String(bytes));
+                Element row = html.getElementById("drop_1");
+                Elements columns = row.getElementsByTag("option");
+                if (columns.get(0).text().equals("Course")) {
+                    for (int j = 1; j < columns.size(); j++) {
+                        String parsedCourse = columns.get(j).attr("value");
+                        coursestemp.add(parsedCourse);
+                    }
+                }
+                Gson gson = new Gson();
+                String coursesjson = gson.toJson(coursestemp);
+                pref.edit().putString("courses", coursesjson).apply();
+                courses = new ArrayList<>();
+                String json = pref.getString("courses", null);
+                Type listType = new TypeToken<ArrayList<String>>() {
+                }.getType();
+                ArrayList<String> course = gson.fromJson(json, listType);
+                courses.addAll(course);
+                if (needupdatecourse) {
+                    setCourseSpinner();
+                }
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, byte[] responseBody, Throwable error) {
+                Utils.showToast(AcademicTimetableActivity.this, "An unexpected error occurred. Please try again later");
+                finish();
+            }
+        });
+    }
+
+    private void getBranches(final int position, final List<String> branchestemp) {
+        AsyncHttpClient client = new AsyncHttpClient();
+        String html = getString(R.string.branch_name_url) + courses.get(position);
+        client.get(html, new AsyncHttpResponseHandler() {
+            @Override
+            public void onSuccess(int i, Header[] headers, byte[] bytes) {
+                Document html = Jsoup.parse(new String(bytes));
+                Element row = html.getElementById("drop_2");
+                Elements columns = row.getElementsByTag("option");
+                for (int j = 1; j < columns.size(); j++) {
+                    String parsedCourse = columns.get(j).text().replace(".", "");
```

Comment:
Why using text yet ? Use `attr` and get the value directly instead of replacing dots.

Reply:
Will remove the .replace() alone